src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass_Switch.cs

```diff
@@ -265,5 +266,80 @@ protected virtual void VisitPatternSwitchSection(BoundPatternSwitchSection node,
         }
 
         #endregion implementation for the pattern-matching variation of the switch statement.
+
+        #region implementation for the new recursive pattern-matching variation of the switch statement.
+
+        public override BoundNode VisitPatternSwitchStatement2(BoundPatternSwitchStatement2 node)
+        {
+            // visit switch header
+            LocalState breakState = VisitPatternSwitchHeader2(node);
+
+            // visit switch block
+            VisitPatternSwitchBlock2(node);
+
+            ResolveBreaks(breakState, node.BreakLabel);
+            return null;
+        }
+
+        private void VisitPatternSwitchBlock2(BoundPatternSwitchStatement2 node)
+        {
+            var initialState = State.Clone();
+
+            // PROTOTYPE(patterns2): when the input is a constant, we should simulate dispatch to see if
+            // a particular case is the only reachable one. We will need a spec for reachability to justify it.
+            foreach (var section in node.SwitchSections)
+            {
+                foreach (var label in section.SwitchLabels)
+                {
+                    // We treat all labels as reachable, even if they are subsumed or erroneous.
+                    SetState(initialState.Clone());
+                    VisitPattern(null, label.Pattern);
+                    SetState(StateWhenTrue);
+                    if (label.Guard != null)
+                    {
+                        VisitCondition(label.Guard);
+                        SetState(StateWhenTrue);
+                    }
+                    _pendingBranches.Add(new PendingBranch(label, this.State));
+                }
+            }
+
+            // visit switch sections
+            var afterSwitchState = UnreachableState();
+            var switchSections = node.SwitchSections;
+            var iLastSection = (switchSections.Length - 1);
+            for (var iSection = 0; iSection <= iLastSection; iSection++)
+            {
+                VisitPatternSwitchSection(switchSections[iSection], node.Expression, iSection == iLastSection);
```

Comment:
>VisitPatternSwitchSection [](start = 16, length = 25)

Should this be ```VisitPatternSwitchSection2```?

Reply:
Addressed in https://github.com/dotnet/roslyn/pull/24059/commits/114a56e9077ed03f08443889b69cae6972a4acc6 of https://github.com/dotnet/roslyn/pull/24059